### PR TITLE
shuffle among answers feature added with valid correct answer

### PR DESCRIPTION
**Pull request by Shariar-Hasan**
**`if you merge this, can you please add hacktoberfest-accepted label while merging.`**

- closes: https://github.com/wingkwong/react-quiz-component/issues/88

i have added the feature of

- shuffling among answers , 
- correct answer from shuffled array
some screenshots of same question:
image
---
image
---
image

### CODE REVIEW

File: src/lib/Quiz.jsx
Comment (wingkwong) [inline src/lib/Quiz.jsx:89]: why comment these?

File: src/lib/Quiz.jsx
Comment (wingkwong) [inline src/lib/Quiz.jsx:106]: remove this line

File: src/lib/Quiz.jsx
Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:89]: oh wth, sorry, that was suppose to be uncommented, i was testing it. my mistake. 😥

**Comment by Shariar-Hasan** [comment]
> 1. Just FYI. This repository has been labelled as `hacktoberfest` so that all merged PRs will be counted without adding `hacktoberfest-accepted`. It's stated in the official website.
> 2. Please run the lint (`npm run lint`) to avoid unrelated changes
> 3. It should be more configurable. Let's have a new prop called `shuffleAnswer` so that users could shuffle on demand. See `shuffle` as a reference.
> 4. Update README since there is a new prop.

got it, i am working on it

File: src/lib/Quiz.jsx
Comment (wingkwong) [reply src/lib/Quiz.jsx:106]: L106. 

```
console.log({ q: questions[i] });
```

Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:106]: oh i see, i got it ⭐

File: src/lib/Quiz.jsx
Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:89]: actually now i remembered, i have done it because, as we see, in the previous line, in if and else state, the state of the question have been set, but as the state will not update immediately, so the when the below line(which i have commented) get executed, the answer (i have shufled ) will not be there.
if you need questionIndex in each question, i can do this in another way, what do you say?

image

Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:89]: image
i have still commented out the same line, and create some changes inside the if and else above the commented line.
is it okay?

**Comment by Shariar-Hasan** [comment]
dear @wingkwong ,
first of all, sorry for asking many questions, as i am a newbie in PR related things,

i have created a PR, can you verify now.
i tried to do what ever you asked for,
only one thing is giving me error, which is this 
image

so the recent pull request is okay or i have to makje some other changes?

File: src/lib/Quiz.jsx
Comment (wingkwong) [reply src/lib/Quiz.jsx:104]: Oh ok. that one seems unnecessary. Your approach is fine.

Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:101]: Thank you brother❤️

File: src/lib/Quiz.jsx
Comment (wingkwong) [inline src/lib/Quiz.jsx:69]: shuffleAnswer shouldn't depend on shuffleQuestion. In other words, I should be able to just shuffle answer without shuffling questions. This shouldn't placed inside `shuffleQuestions`

**Review by wingkwong — CHANGES_REQUESTED** [review]
Overall it looks good. Just one more change. I've cleaned up a bit. Please pull the code first.

**Comment by Shariar-Hasan** [comment]
image
brohter,
they didnt count my contribution. it says, the maintainer must add hacktoberfest-accepted label on pull request label 😓😓

File: src/lib/Quiz.jsx
Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:69]: ok, i got it, let me do it

**Comment by wingkwong** [comment]
> brohter, they didnt count my contribution. it says, the maintainer must add hacktoberfest-accepted label on pull request label 😓😓

It is stated clearly _To be accepted, a maintainer either needs to **MERGE** your PR, ..._.  and so far this PR is not ready to merge yet. You can't expect people just accept everything you made even it is not working. Hence, it is still not accepted. Once you fix that problem, I'll merge it and you will see it is accepted. Just be patient.

**Comment by Shariar-Hasan** [comment]
> > brohter, they didnt count my contribution. it says, the maintainer must add hacktoberfest-accepted label on pull request label 😓😓
> 
> It is stated clearly _To be accepted, a maintainer either needs to **MERGE** your PR, ..._. and so far this PR is not ready to merge yet. You can't expect people just accept everything you made even it is not working. Hence, it is still not accepted YET. Once you fix that problem, I'll merge it and you will see it is accepted. Just be patient.

i am extremely sorry for the misunderstanding, as i just woke up, just saw this

image

i thought the PR got merged .
its my mistake, i appologise

**Comment by Shariar-Hasan** [comment]
hi @wingkwong  i have commited some changes, can you review?

**Comment by wingkwong** [comment]
Next time you may click re-request review here.

image

**Review by wingkwong — CHANGES_REQUESTED** [review]
I can't even start the app.

File: src/lib/Quiz.jsx
Comment (wingkwong) [inline src/lib/Quiz.jsx:101]: merge conflict here. please fix that.

File: src/lib/Quiz.jsx
Comment (wingkwong) [inline src/lib/Quiz.jsx:72]: remove the code if not in use.

File: src/lib/Quiz.jsx
Comment (wingkwong) [inline src/lib/Quiz.jsx:84]: remove these comments - already self-explanatory.

File: src/lib/Quiz.jsx
Comment (wingkwong) [inline src/lib/Quiz.jsx:89]: same as above.

File: src/lib/Quiz.jsx
Comment (wingkwong) [inline src/lib/Quiz.jsx:94]: dont understand what it means by `(used min function for if )`

Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:94]: at first i tried to do this
image
then i realised you have already validate the "nrOfQuestions" variable above.
so i deleted the Math.min section and kept as it was, but forgot to delete the comment

File: src/lib/Quiz.jsx
Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:72]: ok got it

File: src/lib/Quiz.jsx
Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx:84]: ok got it

File: src/lib/Quiz.jsx
Comment (Shariar-Hasan) [reply src/lib/Quiz.jsx]: send a PR,
i cant figure out how to fix this,
i had to ctrl+z (undo) many thing, and then tried again.
i hope now it will work 😔😔😔

**Comment by wingkwong** [comment]
Overall it looks fine. I'll have a final review tmr.

**Comment by Shariar-Hasan** [comment]
ok 😀 finally its done i hope.

**Review by wingkwong — APPROVED** [review]

**Pull request merged**
